erpnext/accounts/report/balance_sheet/balance_sheet.py

```diff
@@ -31,9 +31,26 @@ def execute(filters=None):
 	message, opening_balance = check_opening_balance(asset, liability, equity)
 
 	data = []
+	#add assets to the balance sheet report
 	data.extend(asset or [])
+
+	#add liabilities to the balance sheet report
 	data.extend(liability or [])
-	data.extend(equity or [])
+
+	#add provisional profit/loss and adjust equity totals if books are not closed
+	if len(equity) > 2 and provisional_profit_loss:
+		equity = insert_equity_provisions(equity, provisional_profit_loss, period_list)
+		data.extend(equity or [])
+
+	#specific case when equity is completely empty but we still need to add
+	elif not equity and provisional_profit_loss:
+		equity = create_equity_provisions(provisional_profit_loss, period_list)
```

Comment:
- Resolve the naming issue.
- Insert and create means the same as it implies that you'll end up inserting data in to db by creating it

Reply:
resolved.